content/cpp/concepts/sets/terms/find/find.md

```diff
@@ -0,0 +1,178 @@
+## Syntax 
+
+```pseudo 
+std::set<data_type>::iterator set_name.find(const data_type &value); 
+``` 
+
+**Parameters:** 
+
+- **`set_name`**: The name of the `std::set` being searched. 
+- **`value`**: The element to search for in the set. 
+
+**Return Value:** 
+
+- If the element is found, it returns an iterator pointing to the element. 
+- If the element is not found, it returns an iterator to `set.end()`. 
+
+Before diving into real-world applications, let's start with a generic example that demonstrates the fundamental usage of `.find()` in a `std::set`. 
+
+## Example 1 - How to find an element in an array using `.find()` 
+
+The following example demonstrates how to search for an element in a `std::set` and determine whether it exists. 
+
+ ```cpp
+#include <iostream> 
+#include <set> 
+ 
+int main() { 
+    std::set<int> numbers = {10, 20, 30, 40, 50}; 
+    int searchValue = 30; 
+     
+    auto it = numbers.find(searchValue); 
+     
+    if (it != numbers.end()) { 
+        std::cout << "Element found: " << *it << std::endl; 
+    } else { 
+        std::cout << "Element not found" << std::endl; 
+    } 
+     
+    return 0; 
+}
+```
+  
+**Output** 
+
+```shell  
+Element found: 30 
+``` 
+
+Now that we've seen a basic example, let's explore practical use cases of  `.find()`. 
+
+## Example 2 - Checking if a Product Exists in Inventory 
+
+This example shows how `.find()` helps determine whether a product ID exists in the inventory. If found, the system confirms the availability otherwise, it states that the product is unavailable. 
+
+```cpp 
+#include <iostream> 
+#include <set> 
+
+int main() { 
+
+    std::set<int> productIDs = {101, 202, 303, 404, 505}; 
+    int searchID = 303; 
+    std::set<int>::iterator it = productIDs.find(searchID); 
+
+    if (it != productIDs.end()) { 
+
+        std::cout << "Product ID " << searchID << " is available in inventory." << std::endl; 
+
+    } else { 
+
+        std::cout << "Product ID " << searchID << " is not in inventory." << std::endl; 
+
+    } 
+
+    return 0; 
+
+} 
+``` 
+
+### Output 
+
+```shell 
+Product ID 303 is available in inventory. 
+``` 
+
+## Example 3 - Searching for Employees in a Company Directory 
+
+This example uses `.find()` to search for an employee’s name in a directory, ensuring efficient access to records. 
+
+```cpp 
+#include <iostream> 
+#include <set> 
+#include <string> 
+
+struct CompareIgnoreCase { 
+
+    bool operator()(const std::string &a, const std::string &b) const { 
+        return a < b; 
+
+    } 
+
+}; 
+
+int main() { 
+
+    std::set<std::string, CompareIgnoreCase> names = {"Alice", "Bob", "Charlie"}; 
+    std::string searchName = "Bob"; 
+    auto it = names.find(searchName);   
+
+    if (it != names.end()) { 
+
+        std::cout << "Name found: " << *it << std::endl; 
+
+    } else { 
+
+        std::cout << "Name not found." << std::endl; 
+
+    } 
+
+    return 0; 
+
+} 
+``` 
+
+### Output 
+
+```shell 
+Name found: Bob 
+``` 
```

Comment:
Same as above comment - Please make this part uniform as the above example, preferred format would be to keep it as `**Output**` or to just write it as a sentence like - `The output will be:`.